src/plugins/gs-flatpak.c

```diff
@@ -2329,11 +2349,26 @@ gs_flatpak_app_install (GsFlatpak *self,
 			return FALSE;
 		}
 		g_debug ("installing bundle %s", gs_app_get_unique_id (app));
-		xref = flatpak_installation_install_bundle (self->installation,
-							    gs_app_get_local_file (app),
-							    gs_flatpak_progress_cb,
-							    app,
-							    cancellable, error);
+
+		while (TRUE) {
+			xref = flatpak_installation_install_bundle (self->installation,
+								    gs_app_get_local_file (app),
+								    gs_flatpak_progress_cb,
+								    app,
+								    cancellable, error);
+			if (xref != NULL || !keep_trying)
+				break;
+
+			keep_trying = FALSE;
+
+			if (xref == NULL) {
```

Comment:
If _xref_ were not NULL then it would never reach this block, right? So we can drop the check.

I'd probably just use this block with _&& keep_trying_, and set _keep_trying_ inside of it (makes the code more clear IMO).